File: gen_projects.js
const fs = require('fs');
const path = require('path');

const baseDir = "c:\\Users\\HARMONY\\Desktop\\portfolioarchi-main\\src\\assets";
const categories = {
    "COMMERCE": "COMMERCE",
    "HOTEL": "HÔTEL",
    "RESIDENCE": "RÉSIDENCE",
    "BATIMENT MULTIFONCTIONNEL": "BÂTIMENT MULTIFONCTIONNEL",
    "UNIVERSITAIRE": "UNIVERSITAIRE"
};

function getFiles(catFolder) {
    const fullPath = path.join(baseDir, catFolder);
    let allFiles = [];
    if (!fs.existsSync(fullPath)) return [];

    function walkSync(currentDir) {
        fs.readdirSync(currentDir).forEach(file => {
            const filePath = path.join(currentDir, file);
            if (fs.statSync(filePath).isDirectory()) {
                walkSync(filePath);
            } else if (file.toLowerCase().endsWith('.jpg') || file.toLowerCase().endsWith('.png') || file.toLowerCase().endsWith('.jpeg')) {
                const relPath = path.relative(baseDir, filePath);
                const encodedPath = relPath.split(path.sep).map(p => encodeURIComponent(p)).join('/');
                allFiles.push({
                    url: `assets/${encodedPath}`,
                    caption: file.split('.')[0].replace(/_/g, ' ').replace(/-/g, ' ').trim()
                });
            }
        });
    }
    walkSync(fullPath);
    return allFiles;
}

let tsCode = "  projects: ProjectGroup[] = [\n";
Object.entries(categories).forEach(([folder, label]) => {
    const images = getFiles(folder);
    const catId = label.toLowerCase().replace(/ /g, '-').replace(/ô/g, 'o').replace(/é/g, 'e').replace(/â/g, 'a');
    
    tsCode += `    // ─── ${label} ──────────────────────────────────────────────────────────\n`;
    tsCode += `    {\n`;
    tsCode += `      id: '${catId}',\n`;
    tsCode += `      tag: '${label}',\n`;
    tsCode += `      title: '${label === 'BATIMENT MULTIFONCTIONNEL' ? 'Bâtiments Multifonctionnels' : label}',\n`;
    tsCode += `      description: \`<p>Cet ensemble regroupe l'intégralité des projets et études réalisés dans le domaine : <strong>${label.toLowerCase()}</strong>.</p><p>Découvrez nos rendus 3D, plans de masse et photographies de chantiers.</p>\`,\n`;
    tsCode += `      tech: ['ArchiCAD', 'Lumion', 'Conception', 'Suivi de chantier'],\n`;
    tsCode += `      images: [\n`;
    images.forEach(img => {
        tsCode += `        { url: '${img.url}', caption: '${img.caption}' },\n`;
    });
    tsCode += `      ]\n`;
    tsCode += `    },\n`;
});
tsCode += "  ];";

console.log(tsCode);

File: gen_projects.py
import os
import json
from urllib.parse import quote

base_dir = r"c:\Users\HARMONY\Desktop\portfolioarchi-main\src\assets"
categories = {
    "COMMERCE": "COMMERCE",
    "HOTEL": "HÔTEL",
    "RESIDENCE": "RÉSIDENCE",
    "BATIMENT MULTIFONCTIONNEL": "BÂTIMENT MULTIFONCTIONNEL",
    "UNIVERSITAIRE": "UNIVERSITAIRE"
}

def get_files(cat_folder):
    full_path = os.path.join(base_dir, cat_folder)
    all_files = []
    if not os.path.exists(full_path):
        return []
    for root, dirs, files in os.walk(full_path):
        for file in files:
            if file.lower().endswith(('.jpg', '.png', '.jpeg')):
                rel_path = os.path.relpath(os.path.join(root, file), base_dir)
                # URL encode the path but keep / as /
                encoded_path = "/".join([quote(p) for p in rel_path.split(os.sep)])
                all_files.append({
                    "url": f"assets/{encoded_path}",
                    "caption": file.split('.')[0].replace('_', ' ').replace('-', ' ').strip()
                })
    return all_files

results = {}
for folder, label in categories.items():
    images = get_files(folder)
    results[label] = images

# Output as TS-like string
ts_code = "  projects: ProjectGroup[] = [\n"
for label, images in results.items():
    cat_id = label.lower().replace(' ', '-').replace('ô', 'o').replace('é', 'e').replace('â', 'a')
    ts_code += f"    // ─── {label} ──────────────────────────────────────────────────────────\n"
    ts_code += "    {\n"
    ts_code += f"      id: '{cat_id}',\n"
    ts_code += f"      tag: '{label}',\n"
    ts_code += f"      title: '{label.title() if 'MULTIFONCTIONNEL' not in label else 'Bâtiments Multifonctionnels'}',\n"
    ts_code += f"      description: '<p>Sélection de projets dans le domaine : {label.lower()}.</p>',\n"
    ts_code += "      tech: ['ArchiCAD', 'Lumion', 'Conception'],\n"
    ts_code += "      images: [\n"
    for img in images:
        ts_code += f"        {{ url: '{img['url']}', caption: '{img['caption']}' }},\n"
    ts_code += "      ]\n"
    ts_code += "    },\n"
ts_code += "  ];"

print(ts_code)
